pyccel/ast/core.py

```diff
@@ -5177,15 +5177,61 @@ def bodies(self):
         return b
 
 
-class IfTernaryOperator(If):
+class IfTernaryOperator(Basic, PyccelAstNode):
+    """Represents a IfTernaryOperator statement in the code.
 
-    """class for the Ternery operator"""
-    def __init__(self, *args):
-        for arg in self.args:
-            if len(arg[1].body)!=1:
-                raise TypeError('IfTernary body must be of length 1')
+    Parameters
+    ----------
+    args :
+        args : type list
+        format : condition , value_if_true, value_if_false
+
+    Examples
+    --------
+    >>> from sympy import Symbol
+    >>> from pyccel.ast.core import Assign, IfTernaryOperator
+    >>> n = Symbol('n')
+    >>> x = 5 if n > 1 else 2
+    >>> IfTernaryOperator(PyccelGt(n > 1),  5,  2)
+    IfTernaryOperator(PyccelGt(n > 1),  5,  2)
+    """
+    def __init__(self, cond, value_true, value_false):
+        self._cond = cond
+        self._value_true = value_true
+        self._value_false = value_false
+
+        if self.stage == 'syntactic':
+            return
+        if isinstance(value_true , Nil) or isinstance(value_false, Nil):
+            raise NotImplementedError('None is not implemeted for Ternary Operator')
+        if isinstance(value_true.dtype, NativeString) ^ isinstance(value_false.dtype, NativeString):
+            raise TypeError('Only one of the Ternary Operator results is type string')
```

Comment:
Here it might be better to actually use `errors.raise` instead of a `NotImplementedError` as this is not something that we could implement in the future but is actually a pyccel restriction

Reply:
E.g https://github.com/pyccel/pyccel/blob/master/pyccel/ast/core.py#L223